worker/cluster_worker.ml

```diff
@@ -236,6 +236,13 @@ let rec maybe_prune t queue =
       (fun () ->
          Lwt_process.exec ("", [| "docker"; "system"; "prune"; "-af" |]) >>= function
          | Unix.WEXITED 0 ->
+           (match obuilder with
+           | None -> Lwt.return 0
+           | Some obuilder ->
+             match Obuilder_build.backend obuilder with
+             | `Native _ -> Lwt.return 0
+             | `Docker _ -> Obuilder_build.purge obuilder) >>= fun n ->
```

Comment:
Given that we are just ignoring the values wrapped in the `Native | Docker` tags here, why do we need to expose them in the `Boulder_build` API at all?